src/components/GitmojiList/index.js

```diff
@@ -32,6 +36,39 @@ const GitmojiList = (props: Props): Element<'div'> => {
       })
     : props.gitmojis
 
+  const addPinned = (code: string): void => {
+    setPinneds([...pinneds, code])
+    notify(
+      `<p>Gitmoji <span class="gitmoji-code">${code}</span> pinned to the top 😜</p>`
+    )
+  }
+
+  const removePinned = (code: string): void => {
+    setPinneds(pinneds.filter((pinned) => pinned !== code))
+    notify(
+      `<p>Gitmoji <span class="gitmoji-code">${code}</span> is unpinned 😜</p>`
+    )
+  }
+
+  const isPinned = (code: string): boolean => {
+    return pinneds.includes(code)
+  }
+
+  const sortPinnedGitmojis = (gitmojis: Array<Gitmojis>): Array<Gitmojis> => {
+    return gitmojis.sort((gitmoji) => {
+      if (isPinned(gitmoji.code)) return -1
+      return 0
+    })
+  }
```

Comment:
```suggestion
  const pinDictionary = Object.fromEntries(pinneds.map(code => [code, -1]));

  const sortPinnedGitmojis = (gitmojis: Array<Gitmojis>): Array<Gitmojis> => {
    return gitmojis.sort((gitmoji) => {
      return pinDictionary[gitmoji.code] ?? 0;
    })
  }
```
less array scans :D if the nullish coalescing operator (`??`) is not available, in this case we can just replace it with an OR (`||`)

Reply:
Not a real fan of micro optimisation but by using a `Set` with the `.has()` method it increases performance and still keep the code clear.

Reply:
oh yeah, nice :D, in paper I also know that micro optimisation is bad, but I just can't help myself ahah